docs/paper/dev/api/scheduler.md

```diff
@@ -0,0 +1,197 @@
+`TimeUnit.SECONDS.toMinutes(ticks / 20)`
+
+## Obtaining the scheduler
+
+To obtain a scheduler you can use the instance method on the `Server` class, e.g. in your `onEnable` method:
+
+```java
+@Override
+public void onEnable() {
+	BukkitScheduler scheduler = this.getServer().getScheduler();
+}
+```
+
+## Scheduling tasks
+
+Scheduling a task requires you to pass:
+- your plugin's instance,
+- the code to run, either with a `Runnable` or `Consumer<BukkitTask>`,
+- the delay in ticks before the task should run,
+- if you're scheduling a repeating task - the period in ticks between each execution of the task.
+
+### Difference between synchronous and asynchronous tasks
+
+#### Synchronous tasks (on the main thread)
+
+Synchronous tasks are tasks that are executed on the main server thread. This is the same
+thread that handles all game logic.
+
+All tasks scheduled on the main thread will affect the server's performance. If your task
+is making web requests, accessing files, databases or otherwise time-consuming operations you should consider using 
+an asynchronous task.
+
+#### Asynchronous tasks (off the main thread)
+
+Asynchronous tasks are tasks that are executed on separate threads, therefore will not affect
+your server's performance.
+
+:::warning
+
+**You cannot safely access the Bukkit API from within asynchronous tasks**. If a method is not explicitly marked
+as thread-safe or isn't obviously designed to be used asynchronously **you will eventually encounter errors**.
+
+:::
+
+:::info
+
+While the tasks are executed on separate threads, they are still started from the main thread
+and will be affected if the server is lagging, an example would be 20 ticks not being exactly 1 second.
+
+If you need a scheduler that runs independently of the server consider using your own
+[`ScheduledExecutorService`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/ScheduledExecutorService.html).
+You can follow [this guide](https://www.baeldung.com/java-executor-service-tutorial#ScheduledExecutorService) to learn how to use it.
+
+:::
+
+### Difference between `Runnable` and `Consumer<BukkitTask>`
+
+#### Using `Runnable`
+
+The `Runnable` interface is used for the simplest tasks that don't require a `BukkitTask` instance.
+
+You can either implement it in a separate class, e.g.:
+
+```java
+public class MyRunnableTask implements Runnable {
+
+	private final MyPlugin plugin;
+	
+	public MyRunnableTask(MyPlugin plugin) {
+		this.plugin = plugin;
+	}
+	
+	@Override
+	public void run() {
+		this.plugin.getServer().broadcast(Component.text("Hello, World!"));
+	}
+
+}
+```
+```java
+scheduler.runTaskLater(plugin, new MyRunnableTask(plugin), 20);
+```
+
+Or use a lambda expression which is great for simple and short tasks:
+
+```java
+scheduler.runTaskLater(
+	plugin, 
+	// Lambda:
+	() -> {
+		this.plugin.getServer().broadcast(Component.text("Hello, World!"));
+	},
+	// End of the lambda
+	20);
```

Comment:
```suggestion
	plugin, /* Lambda: */ () -> {
		this.plugin.getServer().broadcast(Component.text("Hello, World!"));
	}, /* End of the lambda */ 20);
```

Keeps it styled the same as the next example and looks cleaner

Reply:
Will change